SurveyApp/Extensions/UIKit/UIViewController/UIViewController+Error.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  UIViewController+Error.swift
+//  SurveyApp
+//
+//  Created by Chananchida F. on 6/18/21.
+//
+
+import UIKit
+
+protocol ViewInput {
+    
+    func showError(message: String)
+}
+
+extension ViewInput where Self: UIViewController {
```

Comment:
Imagine we have Alert, Confirm, and ShowProgressHUD and
-  `ViewControllerA` only need Alert, ShowProgressHUD
- `ViewControllerB` only need Alert
- `ViewControllerC` only need Confirm, ShowProgressHUD

So if we follow your current way, all A, B, C will be included what they don't need 
My suggestion here that we should separate these and adapt only what the controller needs 😁